sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,301 @@
+    # assert child_joint_point.pos_from(parent_joint_point) == dis * parent.frame.x
+    child_joint_masscenter = l * (child.frame.x + child.frame.y)
+    assert child_joint_point.pos_from(child.masscenter) == child_joint_masscenter
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+    assert child.frame.ang_vel_in(parent.frame) == omega * parent.frame.x
+
+
+def test_cylindrical_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    cylindrical_joint = CylindricalJoint('cylindrical_joint', parent, child,
+                                         parent_point_pos=(0, l1, 0),
+                                         child_point_pos=(0, l2, 0),
+                                         parent_axis=parent.frame.y,
+                                         child_axis=parent.frame.z)
+    assert cylindrical_joint.name == 'cylindrical_joint'
+    assert cylindrical_joint.parent == parent
+    assert cylindrical_joint.child == child
+    assert cylindrical_joint.parent_axis == parent.frame.y
+    assert cylindrical_joint.child_axis == parent.frame.z
+    assert cylindrical_joint._parent_joint_location == l1 * parent.frame.y
+    assert cylindrical_joint._child_joint_location == l2 * child.frame.y
+
+
+def test_planar_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    planar_joint = PlanarJoint('planar_joint', parent, child,
+                               child_point_pos=(0, l, 0),
+                               parent_axis=parent.frame.y,
+                               child_axis=child.frame.y)
+
+    assert planar_joint.name == 'planar_joint'
+    assert planar_joint.parent == parent
+    assert planar_joint.child == child
+    assert planar_joint._parent_joint_location == Vector(0)
+    assert planar_joint._child_joint_location == l * child.frame.y
+    assert planar_joint.parent_axis == parent.frame.y
+
+    theta = dynamicsymbols('planar_joint_theta')  # rotation around z axis.
+    thetad = dynamicsymbols('planar_joint_theta', 1)
+    omega = dynamicsymbols('planar_joint_omega')
+    disx = dynamicsymbols('planar_joint_disx')  # translation along x axis.
+    disxd = dynamicsymbols('planar_joint_disx', 1)
+    velx = dynamicsymbols('planar_joint_velx')
+    disy = dynamicsymbols('planar_joint_disy')  # translation along y axis.
+    disyd = dynamicsymbols('planar_joint_disy', 1)
+    vely = dynamicsymbols('planar_joint_vely')
+
+    assert planar_joint.coordinates == [theta, disx, disy]
+    assert planar_joint.speeds == [omega, velx, vely]
+    assert planar_joint.kds == [thetad - omega, disxd - velx, disyd - vely]
+
+    child_joint_point = planar_joint.child_joint_point
+    parent_joint_point = planar_joint.parent_joint_point
+    assert parent_joint_point.name == 'planar_joint_parent_joint'
+    assert child_joint_point.name == 'planar_joint_child_joint'
```

Comment:
This is how I imagine a planar joint:

1. Two bodies with center of masses and body fixed reference frames.
2. Create a body fixed plane on each body that is a distance "d" from the mass center of each respective body. This only needs two distances and two normal vectors.

There is no single point involved, so it is odd to think about. You could say that there are points in the planes that are the projections of the centers of mass. But it is odd.